scripts/meta/west/cmake.py

```diff
@@ -0,0 +1,206 @@
+# Copyright (c) 2018 Open Source Foundries Limited.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+'''Helpers for dealing with CMake'''
+
+from collections import OrderedDict
+import re
+import subprocess
+import shutil
+
+from . import log
+from .util import quote_sh_list
+
+__all__ = ['run_build', 'make_c_identifier', 'CMakeCacheEntry', 'CMakeCache']
+
+DEFAULT_CACHE = 'CMakeCache.txt'
+
+
+def run_build(directory, extra_args=[], quiet=False):
```

Comment:
'directory' is ambiguous.

It should be renamed to build_directory

or removed and leave it to the invoker to specify the build directory through extra_args.

Reply:
> It should be renamed to build_directory

I'm fine with that.

> or removed and leave it to the invoker to specify the build directory through extra_args.

I don't think this would be a good change. CMake run in build tool mode must have `--build BUILD_DIR` as the first argument. Since the purpose of this function is to run in build tool mode, having the build directory as the first mandatory positional argument makes sense.